velox/exec/Task.cpp

```diff
@@ -792,34 +792,33 @@ std::unique_ptr<ContinuePromise> Task::addSplitLocked(
         not split.hasGroup(), "Got split group for ungrouped execution!");
     return addSplitToStoreLocked(
         splitsState.groupSplitsStores[0], std::move(split));
-  } else {
-    VELOX_CHECK(split.hasGroup(), "Missing split group for grouped execution!");
-    const auto splitGroupId = split.groupId; // Avoid eval order c++ warning.
-    // If this is the 1st split from this group, add the split group to queue.
-    // Also add that split group to the set of 'seen' split groups.
-    if (seenSplitGroups_.find(splitGroupId) == seenSplitGroups_.end()) {
-      seenSplitGroups_.emplace(splitGroupId);
-      queuedSplitGroups_.push(splitGroupId);
-      auto self = shared_from_this();
-      // We might have some free driver slots to process this split group.
-      ensureSplitGroupsAreBeingProcessedLocked(self);
-    }
-    return addSplitToStoreLocked(
-        splitsState.groupSplitsStores[splitGroupId], std::move(split));
   }
+  VELOX_CHECK(split.hasGroup(), "Missing split group for grouped execution!");
```

Comment:
nit: add empty line before this statement

Should this be a USER_CHECK? Perhaps, add task ID to the error message.